WordPress/src/main/java/org/wordpress/android/ui/main/utils/MainCreateSheetHelper.kt

```diff
@@ -0,0 +1,92 @@
+package org.wordpress.android.ui.main.utils
+
+import org.wordpress.android.analytics.AnalyticsTracker.Stat
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.bloggingprompts.BloggingPromptsSettingsHelper
+import org.wordpress.android.ui.main.MainActionListItem
+import org.wordpress.android.ui.main.WPMainNavigationView.PageType
+import org.wordpress.android.ui.mysite.cards.dashboard.bloggingprompts.BloggingPromptAttribution
+import org.wordpress.android.ui.voicetocontent.VoiceToContentFeatureUtils
+import org.wordpress.android.util.BuildConfigWrapper
+import org.wordpress.android.util.SiteUtilsWrapper
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.util.config.ReaderFloatingButtonFeatureConfig
+import java.util.Locale
+import javax.inject.Inject
+
+class MainCreateSheetHelper @Inject constructor(
+    private val voiceToContentFeatureUtils: VoiceToContentFeatureUtils,
+    private val readerFloatingButtonFeatureConfig: ReaderFloatingButtonFeatureConfig,
+    private val bloggingPromptsSettingsHelper: BloggingPromptsSettingsHelper,
+    private val buildConfig: BuildConfigWrapper,
+    private val siteUtils: SiteUtilsWrapper,
+    private val analyticsTracker: AnalyticsTrackerWrapper,
+) {
+    fun shouldShowFabForPage(page: PageType?): Boolean {
+        val enabledForPage = page == PageType.MY_SITE ||
+                (page == PageType.READER && readerFloatingButtonFeatureConfig.isEnabled())
+        return buildConfig.isCreateFabEnabled && enabledForPage
+    }
+
+    @Suppress("FunctionOnlyReturningConstant")
+    fun canCreatePost(): Boolean = true // for completeness
+
+    fun canCreatePage(site: SiteModel?, page: PageType?): Boolean {
+        return siteUtils.hasFullAccessToContent(site) && page == PageType.MY_SITE
+    }
+
+    fun canCreatePostFromAudio(site: SiteModel?): Boolean {
+        return voiceToContentFeatureUtils.isVoiceToContentEnabled() && siteUtils.hasFullAccessToContent(site)
+    }
+
+    suspend fun canCreatePromptAnswer(): Boolean = bloggingPromptsSettingsHelper.shouldShowPromptsFeature()
+
+    // region Analytics
```

Comment:
💡 Perhaps in the future we can split the analytics logic into its own class (e.g. `MainCreateSheetTracker`)

Reply:
Done in PR targeting `release`